third_party/closure-compiler/runner/src/AmpPass.java

```diff
@@ -0,0 +1,67 @@
+import com.google.javascript.jscomp.AbstractCompiler;
+import com.google.javascript.jscomp.HotSwapCompilerPass;
+import com.google.javascript.jscomp.NodeTraversal;
+import com.google.javascript.jscomp.NodeTraversal.AbstractPostOrderCallback;
+import com.google.javascript.jscomp.NodeUtil;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+
+class AmpPass extends AbstractPostOrderCallback implements HotSwapCompilerPass {
+  
+  final AbstractCompiler compiler;
+  
+  public AmpPass(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+    hotSwapScript(root, null);
+  }
+
+  @Override
+  public void hotSwapScript(Node scriptRoot, Node originalRoot) {
+    NodeTraversal.traverseEs6(compiler, scriptRoot, this);
+  }
+
+  @Override
+  public void visit(NodeTraversal t, Node n, Node parent) {
+    if (n.isExprResult()) {
+      maybeEliminateExpressionByName(t, n, parent);
+    }
+  }
+ 
+  private void maybeEliminateExpressionByName(NodeTraversal t, Node n, Node parent) {
+    // n = EXPRESSION_RESULT > CALL > GETPROP
+    Node call = n.getFirstChild();
+    if (call == null) {
+      return;
+    }
+    Node expression = call.getFirstChild();
+    if (expression == null) {
+      return;
+    }
+
+    if (qualifiedNameEndsWithStripType(expression)) {
+      if (parent.isExprResult()) {
+        Node grandparent = parent.getParent();
+        grandparent.replaceChild(parent, IR.empty());
+      } else {
+        parent.replaceChild(n, IR.empty());
+      }
+    }
+    compiler.reportCodeChange();
+  }
+  
+  boolean qualifiedNameEndsWithStripType(Node n) {
+    String name = n.getQualifiedName();
+    return qualifiedNameEndsWithStripType(name);
+  }
+
+  boolean qualifiedNameEndsWithStripType(String name) {
+    if (name != null) {
+      return name.endsWith("log.dev.fine");
```

Comment:
@dvoytenko @cramforce this is what i came up with which is based on the closure compilers' `stripTypePrefixes` code, this code is more or less a `stripTypeSuffixes` (which isnt implemented)